web/src/settings_invites.ts

```diff
@@ -325,6 +370,108 @@ export function on_load_success(
 
         $(".dialog_submit_button").attr("data-invite-id", invite_id);
     });
+
+    $(".admin_invites_table").on("click", ".edit-invite", function (this: HTMLElement, e) {
+        // This click event must not get propagated to parent container otherwise the modal
+        // will not show up because of a call to `close_active` in `settings.js`.
+        e.preventDefault();
+        e.stopPropagation();
+
+        const invite_id = $(this).attr("data-invite-id")!;
+        $(".dialog_submit_button").attr("data-invite-id", invite_id);
+        const invite: Invite | undefined = meta.invites.find(
+            (invite) => invite.id === Number.parseInt(invite_id, 10) && invite.is_multiuse,
+        );
+        const html_body = render_edit_invite_user_modal({
+            invite_as_options: settings_config.user_role_values,
+            is_admin: current_user.is_admin,
+            is_owner: current_user.is_owner,
+            show_select_default_streams_option: stream_data.get_default_stream_ids().length > 0,
+            can_subscribe_other_users:
+                settings_data.can_subscribe_others_to_all_accessible_streams(),
+        });
+
+        function edit_invite_user_modal_post_render(): void {
+            const $submit_button = $("#edit-invite-form .dialog_submit_button");
+            $submit_button.prop("disabled", true);
+            $submit_button.attr("data-loading-text", $t({defaultMessage: "Updating link…"}));
+            if (invite?.is_multiuse) {
+                const $stream_pill_container = $("#invite_streams_container .pill-container");
+                if (invite.include_realm_default_subscriptions) {
```

Comment:
This is not possible in UI, but we allow invites to have both `stream_ids` and `include_realm_default_subscriptions` set for an invite. So, not sure how to handle that while editing.